### ISSUE TITLE
Thanks for sharing the updated link, but unfortunately the downloaded file is not working.

### ISSUE THREAD

**Issue by Adnan-CAS** [tianrun-chen/SAM-Adapter-PyTorch#79]
Thanks for sharing the updated link, but unfortunately the downloaded file is not working, probably the file(newly uploaded) is broken, Could you please check it.

_Originally posted by @Adnan-CAS in https://github.com/tianrun-chen/SAM-Adapter-PyTorch/issues/77#issuecomment-2247218049_

**Comment by 77waiwai-Demon** [comment]
> Thanks for sharing the updated link, but unfortunately the downloaded file is not working, probably the file(newly uploaded) is broken, Could you please check it.
> 
> _Originally posted by @Adnan-CAS in [#77 (comment)](https://github.com/tianrun-chen/SAM-Adapter-PyTorch/issues/77#issuecomment-2247218049)_

Is the compressed file corrupted?

**Comment by Adnan-CAS** [comment]
Yes, it is corrupted, not working

**Comment by Adnan-CAS** [comment]
> > Thanks for sharing the updated link, but unfortunately the downloaded file is not working, probably the file(newly uploaded) is broken, Could you please check it.
> > _Originally posted by @Adnan-CAS in [#77 (comment)](https://github.com/tianrun-chen/SAM-Adapter-PyTorch/issues/77#issuecomment-2247218049)_
> 
> Is the compressed file corrupted?

Yes, it is corrupted, not working

**Comment by 77waiwai-Demon** [comment]
> > > Thanks for sharing the updated link, but unfortunately the downloaded file is not working, probably the file(newly uploaded) is broken, Could you please check it.
> > > _Originally posted by @Adnan-CAS in [#77 (comment)](https://github.com/tianrun-chen/SAM-Adapter-PyTorch/issues/77#issuecomment-2247218049)_
> > 
> > 
> > Is the compressed file corrupted?
> 
> Yes, it is corrupted, not working

I have checked, and the compressed file I uploaded is normal. Please check it again.

**Comment by Adnan-CAS** [comment]
> > > > Thanks for sharing the updated link, but unfortunately the downloaded file is not working, probably the file(newly uploaded) is broken, Could you please check it.
> > > > _Originally posted by @Adnan-CAS in [#77 (comment)](https://github.com/tianrun-chen/SAM-Adapter-PyTorch/issues/77#issuecomment-2247218049)_
> > > 
> > > 
> > > Is the compressed file corrupted?
> > 
> > 
> > Yes, it is corrupted, not working
> 
> I have checked, and the compressed file I uploaded is normal. Please check it again.

The file type is(.file) not the .pth or .pt , How to use it? I have tried to change the type to(.pt/.pth) but its not working. Please guide. 
![file](https://github.com/user-attachments/assets/0c9ae836-eaf6-4121-9447-61d50b592ad7)

**Comment by 77waiwai-Demon** [comment]
> > > > > Thanks for sharing the updated link, but unfortunately the downloaded file is not working, probably the file(newly uploaded) is broken, Could you please check it.
> > > > > _Originally posted by @Adnan-CAS in [#77 (comment)](https://github.com/tianrun-chen/SAM-Adapter-PyTorch/issues/77#issuecomment-2247218049)_
> > > > 
> > > > 
> > > > Is the compressed file corrupted?
> > > 
> > > 
> > > Yes, it is corrupted, not working
> > 
> > 
> > I have checked, and the compressed file I uploaded is normal. Please check it again.
> 
> The file type is(.file) not the .pth or .pt , How to use it? I have tried to change the type to(.pt/.pth) but its not working. Please guide. ![file](https://private-user-images.githubusercontent.com/155667335/353295272-0c9ae836-eaf6-4121-9447-61d50b592ad7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIzMDgyNDEsIm5iZiI6MTcyMjMwNzk0MSwicGF0aCI6Ii8xNTU2NjczMzUvMzUzMjk1MjcyLTBjOWFlODM2LWVhZjYtNDEyMS05NDQ3LTYxZDUwYjU5MmFkNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDczMFQwMjUyMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOTk5NzUwNTNhZWFkZTFjNzhlOWJlMTIyMTU5NmNhMWU2NDkxMTY1ZDZhMGNkNWI5OTI0MmM5ZmRhNDJmMTk3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.LO3Vo0f791A14jKfhWpF210TnYs53ddUY4NtkWZKm_4)

I suspect it might be an issue with the extraction tool. You could try using a different extraction tool (for example, Bandzip for Windows), or use the Linux command `tar -xvf` to extract the files.

**Comment by Adnan-CAS** [comment]
> > > > > > Thanks for sharing the updated link, but unfortunately the downloaded file is not working, probably the file(newly uploaded) is broken, Could you please check it.
> > > > > > _Originally posted by @Adnan-CAS in [#77 (comment)](https://github.com/tianrun-chen/SAM-Adapter-PyTorch/issues/77#issuecomment-2247218049)_
> > > > > 
> > > > > 
> > > > > Is the compressed file corrupted?
> > > > 
> > > > 
> > > > Yes, it is corrupted, not working
> > > 
> > > 
> > > I have checked, and the compressed file I uploaded is normal. Please check it again.
> > 
> > 
> > The file type is(.file) not the .pth or .pt , How to use it? I have tried to change the type to(.pt/.pth) but its not working. Please guide. ![file](https://private-user-images.githubusercontent.com/155667335/353295272-0c9ae836-eaf6-4121-9447-61d50b592ad7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIzMDgyNDEsIm5iZiI6MTcyMjMwNzk0MSwicGF0aCI6Ii8xNTU2NjczMzUvMzUzMjk1MjcyLTBjOWFlODM2LWVhZjYtNDEyMS05NDQ3LTYxZDUwYjU5MmFkNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDczMFQwMjUyMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOTk5NzUwNTNhZWFkZTFjNzhlOWJlMTIyMTU5NmNhMWU2NDkxMTY1ZDZhMGNkNWI5OTI0MmM5ZmRhNDJmMTk3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.LO3Vo0f791A14jKfhWpF210TnYs53ddUY4NtkWZKm_4)
> 
> I suspect it might be an issue with the extraction tool. You could try using a different extraction tool (for example, Bandzip for Windows), or use the Linux command `tar -xvf` to extract the files.

I extracted(as suggested) it and got the files but the (Polyp Segmentation - Medical Applications
[Kvasir]) is still missing, as I am working on medical applications so I need that one, Could you please share that one. Thanks

![file2](https://github.com/user-attachments/assets/8ad729ea-169f-4359-b0c8-d34525bdde62)

**Comment by DianiSirimewan** [comment]
@Adnan-CAS Can you please share the configs that you used for training? I am getting this error which I couldn't solve yet.
![Screenshot 2024-11-04 160018](https://github.com/user-attachments/assets/0b47839f-9eb6-4141-ad5e-3cf3619c5387)
These are the configs that I am using.
train_dataset:
  dataset:
    name: paired-image-folders
    args:
      root_path_1: /home/diani/diani/SAM2-Adapter/cod/train/Image
      root_path_2: /home/diani/diani/SAM2-Adapter/cod/train/GT_Instance
      cache: none
      split_key: train
  wrapper:
    name: train
    args:
      inp_size: 1024
      augment: false
  batch_size: 1

val_dataset:
  dataset:
    name: paired-image-folders
    args:
      root_path_1: /home/diani/diani/SAM2-Adapter/cod/train/Image
      root_path_2: /home/diani/diani/SAM2-Adapter/cod/val/GT_Instance
      cache: none
      split_key: test
  wrapper:
    name: val
    args:
      inp_size: 1024
  batch_size: 1

test_dataset:
  dataset:
    name: paired-image-folders
    args:
      root_path_1: /home/diani/diani/SAM2-Adapter/cdw2/test/images
      root_path_2: /home/diani/diani/SAM2-Adapter/cdw2/test/masks
      cache: none
      split_key: test
  wrapper:
    name: val
    args:
      inp_size: 1024
  batch_size: 1

eval_type: cod
sam_checkpoint: /home/diani/diani/SAM2-Adapter/pretrained/sam2.1_hiera_large.pt
data_norm:
  inp:
    sub:
    - 0.5
    div:
    - 0.5
  gt:
    sub:
    - 0.5
    div:
    - 0.5
  gt_rgb:
    sub:
    - 0.5
    div:
    - 0.5
model:
  name: sam
  args:
    inp_size: 1024
    loss: iou
    encoder_mode:
      name: sam
      img_size: 1024
      mlp_ratio: 4
      patch_size: 16
      qkv_bias: true
      use_rel_pos: true
      window_size: 14
      out_chans: 256
      scale_factor: 32
      input_type: fft
      freq_nums: 0.25
      prompt_type: highpass
      prompt_embed_dim: 256
      tuning_stage: 1234
      handcrafted_tune: true
      embedding_tune: true
      adaptor: adaptor
      embed_dim: 1024
      depth: 24
      num_heads: 16
      global_attn_indexes:
      - 5
      - 11
      - 17
      - 23
optimizer:
  name: adamw
  args:
    lr: 0.0002
lr_min: 1.0e-7
epoch_max: 100

multi_step_lr:
  milestones:
  - 1
  gamma: 0.1
epoch_val: 1
epoch_save: 1

#resume: 60
#start_epoch: 60
